### PR TITLE
Join strings instead of adding them.

### PR DESCRIPTION
**Pull request by anntzer**
See e.g. https://docs.python.org/3/library/functions.html#sum that
documents "".join as the preferred method (briefly, because summing (or
repeated addition) has quadratic complexity for strings).

side note:
https://github.com/takluyver/astsearch and https://github.com/hchasestevens/astpath are quite nice for this kind of cleanups...

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
That is exciting that we do not run any of those lines of code in the test suite!

File: lib/matplotlib/backends/backend_pgf.py [hunk 1]
```diff
@@ -582,11 +582,10 @@ def _print_pgf_path_styles(self, gc, rgbFace):
...
         else:
-            dash_str = r"\pgfsetdash{"
-            for dash in dash_list:
-                dash_str += r"{%fpt}" % dash
-            dash_str += r"}{%fpt}" % dash_offset
-            writeln(self.fh, dash_str)
+            writeln(self.fh,
+                    r"\pgfsetdash{"
+                    + "".join(r"{%fpt}" % dash for dash in dash_list)
+                    + "r}{%fpt}" % dash_offset)
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_pgf.py]: The r was a raw string qualifier before and seems to have slipped into the string.

Comment (anntzer) [reply lib/matplotlib/backends/backend_pgf.py]: cf @tacaswell's comment about none of this being tested :p 
fixed

Comment (timhoffm) [reply lib/matplotlib/backends/backend_pgf.py]: On the danger of being pedantic, I would simply write
~~~
            dash_pattern = "".join(r"{%fpt}" % dash for dash in dash_list)
            writeln(self.fh,
                    r"\pgfsetdash{%s}{%fpt}" % (dash_pattern, dash_offset))
~~~

File: lib/matplotlib/backends/backend_pgf.py [hunk 2]
```diff
@@ -582,11 +582,10 @@ def _print_pgf_path_styles(self, gc, rgbFace):
         if dash_list is None:
             writeln(self.fh, r"\pgfsetdash{}{0pt}")
         else:
-            dash_str = r"\pgfsetdash{"
-            for dash in dash_list:
-                dash_str += r"{%fpt}" % dash
-            dash_str += r"}{%fpt}" % dash_offset
-            writeln(self.fh, dash_str)
+            writeln(self.fh,
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_pgf.py]: On the danger of being pedantic, I would simply write
~~~
            dash_pattern = "".join(r"{%fpt}" % dash for dash in dash_list)
            writeln(self.fh,
                    r"\pgfsetdash{%s}{%fpt}" % (dash_pattern, dash_offset))
~~~

Comment (anntzer) [reply lib/matplotlib/backends/backend_pgf.py]: done

**Pull request merged**